InjectionIII/InjectionServer.swift

```diff
@@ -0,0 +1,332 @@
+//
+//  InjectionServer.swift
+//  InjectionIII
+//
+//  Created by John Holdsworth on 06/11/2017.
+//  Copyright © 2017 John Holdsworth. All rights reserved.
+//
+
+let XcodeBundleID = "com.apple.dt.Xcode"
+let injectionQueue = DispatchQueue(label: "InjectionQueue")
+
+var projectInjected = ["": ["": Date.timeIntervalSinceReferenceDate]]
+let MIN_INJECTION_INTERVAL = 1.0
+
+public class InjectionServer: SimpleSocket {
+    var injector: ((_ changed: NSArray) -> Void)? = nil
+    var fileWatchers = [FileWatcher]()
+    var pending = [String]()
+
+    override public class func error(_ message: String) -> Int32 {
+        let saveno = errno
+        DispatchQueue.main.sync {
+            let alert: NSAlert = NSAlert()
+            alert.messageText = "Injection Error"
+            alert.informativeText = String(format:message, strerror(saveno))
+            alert.alertStyle = NSAlert.Style.warning
+            alert.addButton(withTitle: "OK")
+            _ = alert.runModal()
+        }
+        return -1
+    }
+
+    func sendCommand(_ command: InjectionCommand, with string: String?) {
+        writeCommand(command.rawValue, with: string)
+    }
+
+    @objc override public func runInBackground() {
+        write(NSHomeDirectory())
+
+        var candiateProjectFile = appDelegate.selectedProject
+//        var MAS = false
+
+    //    if (!projectFile) {
+    //        XcodeApplication *xcode = (XcodeApplication *)[SBApplication
+    //                           applicationWithBundleIdentifier:XcodeBundleID];
+    //        XcodeWorkspaceDocument *workspace = [xcode activeWorkspaceDocument];
+    //        projectFile = workspace.file.path;
+    //    }
+
+        if candiateProjectFile == nil {
+            DispatchQueue.main.sync {
+                appDelegate.openProject(self)
+            }
+            candiateProjectFile = appDelegate.selectedProject
+//            MAS = true
+        }
+        guard let projectFile = candiateProjectFile else {
+            return
+        }
+
+        NSLog("Connection with project file: \(projectFile)")
+
+        // tell client app the inferred project being watched
+        let key = readString()
+        if key != INJECTION_KEY {
+            return
+        }
+
+        let builder = SwiftEval()
+
+        // client spcific data for building
+        if let frameworks = readString() {
+            builder.frameworks = frameworks
+        } else { return }
```

Comment:
Would it be better to use `guard` instead of `if else` here?

Reply:
Not at all sure